agent/src/util/discovery_operator.rs

```diff
@@ -743,76 +651,49 @@ pub mod start_discovery {
                                                 dh_details.shared,
                                             )
                                             .await?;
-                                        deregistered = discovery_operator
-                                            .mark_offline_or_deregister_discovery_handler(endpoint)
-                                            .await
-                                            .unwrap();
                                     } else {
                                         trace!("do_discover_on_discovery_handler - Discovery Handlers returned error status {}. Marking all instances offline.", status);
                                         // TODO: Possibly mark config as invalid
-                                        // Mark all associated instances as offline by declaring no visible instances
                                         discovery_operator
                                             .update_instance_connectivity_status(
                                                 kube_interface.clone(),
                                                 std::collections::HashMap::new(),
                                                 dh_details.shared,
                                             )
                                             .await?;
-                                        discovery_operator
-                                            .set_discovery_handler_connectivity_status(
-                                                endpoint,
-                                                DiscoveryHandlerStatus::Waiting,
-                                            );
-                                        break;
                                     }
+                                } else {
+                                    return Err(e);
```

Comment:
I'm trying to see if this is a case we want to handle. Can status be empty? Maybe log an error at least?

Reply:
This error is hit when the type of error is not tonic::Status, signaling that it is not a connection error with the Discovery Handler rather one of the Discovery Operator's other functions that errored such as `update_instance_connectivity_status`. This is an error we'd want to bubble up while connection errors can be retried in case the discovery handler comes back online.
Do you think an `error!` log should be omitted before the bubble up?

Reply:
It might help logging from debuggability perspective. I was mainly curious how we want to handle that case and if we should treat it like a connection error.